show_graph.py

```diff
@@ -108,6 +111,70 @@ def AutoLabel(rects, ax):
             ha="center", va="bottom")
 
 
+def DrawTimeSeriesGraph(connection, table_name, y_axis_field, time,
+                        arrangement):
+  """This function draws a time-series graph.
+
+  Args:
+    connection: the connection to get data from the database
+    table_name: the name of the table to get data from
+    y_axis_field: column name in the table for the y axis
+    time: the starting time for the graph
+    arrangement: the type of arrangement in the experiment
+  """
+  def GetListFromDB(time, category, y_axis_field, connection, table_name):
+    condition = ("where time_of_entry >= \"{}\" and"
+                 " category=\"{}\" Group By RunID "
+                 "Order By time_of_entry").format(
+                     time, category)
+    single_list = db_utils.GetFieldFromTable(
+        connection, table_name,
+        field="AVG({}), STDDEV({}), time_of_entry, RunID".format(
+            y_axis_field, y_axis_field),
+        cond=condition)
+    if not single_list:
+      print "Values are not found in table for category {}.".format(
+          category)
+      return None
+
+    return single_list
+
+  direct_list = GetListFromDB(time, "direct-{}".format(arrangement),
+                              y_axis_field, connection, table_name)
+  envoy_list = GetListFromDB(time, "envoy-{}".format(arrangement),
+                             y_axis_field, connection, table_name)
+
+  if direct_list:
+    direct_means = [v[0] for v in direct_list]
+    direct_std = [v[1] for v in direct_list]
+    direct_times = [v[2].time().strftime("%H:%M") if not i % 2 else ""
+                    for i, v in enumerate(direct_list)]
+  else:
+    return
+
+  if envoy_list:
+    envoy_means = [v[0] for v in envoy_list]
+    envoy_std = [v[1] for v in direct_list]
```

Comment:
Here (and above), you can write this more cleanly. Something like: `envoy_means, envoy_std = zip(*v)`